From b1c0c4e368cb1ae565b865edc91adcdb36ddedb5 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Thu, 3 Dec 2020 18:09:24 +0800
Subject: [PATCH] drm/rockchip: vop2: init yuv overlay mode

post bcsh csc module need to know win overlay at yuv or rgb domain.

Change-Id: I246c22ddf0d02f48f515947a48ef058dca36c7a5
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index b4776798a541..175688a35cbc 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -3334,7 +3334,9 @@ static void vop2_crtc_atomic_begin(struct drm_crtc *crtc, struct drm_crtc_state
 	struct vop2_zpos *vop2_zpos;
 	struct vop2_cluster cluster;
 	uint8_t nr_wins = 0;
+	struct rockchip_crtc_state *vcstate = to_rockchip_crtc_state(crtc->state);
 
+	vcstate->yuv_overlay = is_yuv_output(vcstate->bus_format);
 	vop2_zpos = kmalloc_array(vop2->data->win_size, sizeof(*vop2_zpos), GFP_KERNEL);
 	if (!vop2_zpos)
 		return;
-- 
2.35.3

